system_tests/common_test.go

```diff
@@ -60,6 +59,172 @@ import (
 type info = *BlockchainTestInfo
 type client = arbutil.L1Interface
 
+type SecondNodeParams struct {
+	nodeConfig           *arbnode.Config
+	execConfig           *gethexec.Config
+	stackConfig          *node.Config
+	dasConfig            *das.DataAvailabilityConfig
+	initData             *statetransfer.ArbosInitializationInfo
+	useBuilderNodeConfig bool
+}
+
+type TestClient struct {
+	ctx      context.Context
+	Client   *ethclient.Client
+	Backend  *eth.Ethereum
+	Stack    *node.Node
+	Node     *arbnode.Node
+	ExecNode *gethexec.ExecutionNode
+
+	// having cleanup() field makes cleanup customizable from default cleanup methods after calling build
+	cleanup func()
+}
+
+func NewTestClient(ctx context.Context) *TestClient {
+	return &TestClient{ctx: ctx}
+}
+
+func (tc *TestClient) SendSignedTx(t *testing.T, l2Client *ethclient.Client, transaction *types.Transaction, lInfo info) *types.Receipt {
+	return SendSignedTxViaL1(t, tc.ctx, lInfo, tc.Client, l2Client, transaction)
+}
+
+func (tc *TestClient) SendUnsignedTx(t *testing.T, l2Client *ethclient.Client, transaction *types.Transaction, lInfo info) *types.Receipt {
+	return SendUnsignedTxViaL1(t, tc.ctx, lInfo, tc.Client, l2Client, transaction)
+}
+
+func (tc *TestClient) TransferBalance(t *testing.T, from string, to string, amount *big.Int, lInfo info) (*types.Transaction, *types.Receipt) {
+	return TransferBalanceTo(t, from, lInfo.GetAddress(to), amount, lInfo, tc.Client, tc.ctx)
+}
+
+func (tc *TestClient) TransferBalanceTo(t *testing.T, from string, to common.Address, amount *big.Int, lInfo info) (*types.Transaction, *types.Receipt) {
+	return TransferBalanceTo(t, from, to, amount, lInfo, tc.Client, tc.ctx)
+}
+
+func (tc *TestClient) GetBalance(t *testing.T, account common.Address) *big.Int {
+	return GetBalance(t, tc.ctx, tc.Client, account)
+}
+
+func (tc *TestClient) GetBaseFeeAt(t *testing.T, blockNum *big.Int) *big.Int {
+	return GetBaseFeeAt(t, tc.Client, tc.ctx, blockNum)
+}
+
+func (tc *TestClient) SendWaitTestTransactions(t *testing.T, txs []*types.Transaction) {
+	SendWaitTestTransactions(t, tc.ctx, tc.Client, txs)
+}
+
+func (tc *TestClient) DeploySimple(t *testing.T, auth bind.TransactOpts) (common.Address, *mocksgen.Simple) {
+	return deploySimple(t, tc.ctx, auth, tc.Client)
+}
+
+type NodeBuilder struct {
+	// NodeBuilder configuration
+	ctx           context.Context
+	chainConfig   *params.ChainConfig
+	nodeConfig    *arbnode.Config
+	execConfig    *gethexec.Config
+	l1StackConfig *node.Config
+	l2StackConfig *node.Config
+	L1Info        info
+	L2Info        info
+
+	// L1, L2 Node parameters
+	dataDir       string
+	isSequencer   bool
+	takeOwnership bool
+	withL1        bool
```

Comment:
note: in a later PR - we'll also want to incorporate ValidationNode here.. not sure if it'll be in a TestClient or Builder..
What we currently have is a weird/bad workaround.